### ISSUE TITLE
Locked-mode restore doesn't detect removed packages

### ISSUE THREAD

**Issue by bergbria** [NuGet/Home#8262]
## Details about Problem

dotnet.exe --version (if appropriate): 2.2.204 and 3.0.100-preview5-011568

### Automated repro
I've created a self-contained repro. Download [repro_extraStuffInLockFile.zip](https://github.com/NuGet/Home/files/3323115/repro_extraStuffInLockFile.zip) and run repro.bat.

### General details

1. Create a .csproj with a PackageReference
2. `dotnet restore --use-lock-file`
3. Remove the PackageReference
4. `dotnet restore --use-lock-file --locked-mode`

### Expected
The final restore should fail since packages.lock.json references packages that are no longer used.

### Actual
The restore completes without error

**Comment by rrelyea** [comment]
Understand how we should ensure all packages are the correct version.
Don’t know why we would want to protect against removals. 
@anangaur

**Comment by bergbria** [comment]
This isn't a must-have feature. That said, consider this scenario:

1. Person A removes a bunch of packages from a bunch of projects but doesn't submit the lock file changes since the build doesn't enforce it.
2. Person B changes nothing related to packages but does run a restore/build command that causes the lock files to be updated.

It's a bizarre experience for Person B to suddenly have an arbitrary number of edits in his repo which are completely unrelated to his change. It's also bizarre for anyone who reviews the PR. 

It's much preferable for each change to include all related lock file updates. That in turn means it's much preferable for the build system to enforce that behavior.

**Comment by nkolev92** [comment]
Duplicate of https://github.com/NuGet/Home/issues/7930 which was fixed in 5.1.

I think we should definitely honor removals, as the guarantee is repeatability. **All** the same packages.
A package that used to be there, missing now is a failure. 

This should be fixed in NuGet.exe 5.1 and matching dotnet sdks.

I just verified with a nuget.exe built from dev branch (5.2 preview 3+) 
See below:
I update the repro as such so that it runs with a locally built nuget.exe 


NetCoreConsoleApp.csproj.noPackage
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
    <RestoreLockedMode>true</RestoreLockedMode>
  </PropertyGroup>

</Project>
```

NetCoreConsoleApp.csproj.withPackage
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
  </PropertyGroup>

  <ItemGroup>
	<PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
  </ItemGroup>

</Project>
```

```bat
@echo off

setlocal

pushd %~dp0

set localNugetSource=%~dp0localSource
set nugetCache=%~dp0nugetCache
set objDir=%~dp0obj
set projFile=%~dp0foo.csproj

echo ">>> Cleaning/resetting state"
rd /s /q %localNugetSource%
rd /s /q %nugetCache%
rd /s /q %objDir%
del %projFile%
del packages.lock.json

mkdir %localNugetSource%
mkdir %nugetCache%

set NUGET_PACKAGES=%nugetCache%

echo ">>> Perform the initial restore with a PackageReference"
copy NetCoreConsoleApp.csproj.withPackage %projFile%
F:\NuGet.Client\artifacts\VS15\NuGet.exe restore %projFile%

echo ">>> Remove the PackageReference"
del %projFile%
copy NetCoreConsoleApp.csproj.noPackage %projFile%

rd /s /q %objDir%

echo ">>> Perform a locked-mode restore. This should fail saying the lock file is out of date."
F:\NuGet.Client\artifacts\VS15\NuGet.exe restore %projFile%

endlocal
```

The output is the following: 

```cmd
\repro.bat                                                                                                                                                                                                                                                                         ">>> Cleaning/resetting state"
The system cannot find the file specified.
Could Not Find F:\test\8262\repro_extraStuffInLockFile\packages.lock.json
">>> Perform the initial restore with a PackageReference"
        1 file(s) copied.
MSBuild auto-detection: using msbuild version '16.200.19.31805' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\IntPreview\MSBuild\Current\Bin'.
Restoring packages for F:\test\8262\repro_extraStuffInLockFile\foo.csproj...
Committing restore...
Generating MSBuild file F:\test\8262\repro_extraStuffInLockFile\obj\foo.csproj.nuget.g.props.
Generating MSBuild file F:\test\8262\repro_extraStuffInLockFile\obj\foo.csproj.nuget.g.targets.
Writing assets file to disk. Path: F:\test\8262\repro_extraStuffInLockFile\obj\project.assets.json
Writing packages lock file at disk. Path: F:\test\8262\repro_extraStuffInLockFile\packages.lock.json
Restore completed in 500.63 ms for F:\test\8262\repro_extraStuffInLockFile\foo.csproj.

NuGet Config files used:
    F:\test\8262\repro_extraStuffInLockFile\NuGet.Config
    F:\test\NuGet.Config
    C:\Users\nikolev.REDMOND\AppData\Roaming\NuGet\NuGet.Config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

Feeds used:
    https://api.nuget.org/v3/index.json
    https://dotnetfeed.blob.core.windows.net/aspnet-aspnetcore-tooling/index.json
    https://dotnetfeed.blob.core.windows.net/aspnet-aspnetcore/index.json
    https://dotnetfeed.blob.core.windows.net/aspnet-entityframeworkcore/index.json
    https://dotnetfeed.blob.core.windows.net/aspnet-extensions/index.json
    https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
    https://dotnetfeed.blob.core.windows.net/dotnet-windowsdesktop/index.json
">>> Remove the PackageReference"
        1 file(s) copied.
">>> Perform a locked-mode restore. This should fail saying the lock file is out of date."
MSBuild auto-detection: using msbuild version '16.200.19.31805' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\IntPreview\MSBuild\Current\Bin'.
NU1004: The packages lock file is inconsistent with the project dependencies so restore can't be run in locked mode. Please disable RestoreLockedMode MSBuild property or pass explicit --force-evaluate flag to run restore to update the lock file.
Committing restore...
Writing packages lock file at disk. Path: F:\test\8262\repro_extraStuffInLockFile\packages.lock.json
Restore failed in 81.9 ms for F:\test\8262\repro_extraStuffInLockFile\foo.csproj.

NuGet Config files used:
    F:\test\8262\repro_extraStuffInLockFile\NuGet.Config
    F:\test\NuGet.Config
    C:\Users\nikolev.REDMOND\AppData\Roaming\NuGet\NuGet.Config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config

Feeds used:
    https://api.nuget.org/v3/index.json
    https://dotnetfeed.blob.core.windows.net/aspnet-aspnetcore-tooling/index.json
    https://dotnetfeed.blob.core.windows.net/aspnet-aspnetcore/index.json
    https://dotnetfeed.blob.core.windows.net/aspnet-entityframeworkcore/index.json
    https://dotnetfeed.blob.core.windows.net/aspnet-extensions/index.json
    https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
    https://dotnetfeed.blob.core.windows.net/dotnet-windowsdesktop/index.json
```
